src/sdk/sql_cluster_router.cc

```diff
@@ -865,38 +944,58 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::ExecuteSQLBatchReque
 bool SQLClusterRouter::ExecuteInsert(const std::string& db, const std::string& sql, ::hybridse::sdk::Status* status) {
     if (status == NULL) return false;
     std::shared_ptr<::openmldb::nameserver::TableInfo> table_info;
-    DefaultValueMap default_map;
-    uint32_t str_length = 0;
-    if (!GetInsertInfo(db, sql, status, &table_info, &default_map, &str_length)) {
+    std::vector<DefaultValueMap> default_maps;
+    std::vector<uint32_t> str_lengths;
+    if (!GetMultiRowInsertInfo(db, sql, status, &table_info, &default_maps, &str_lengths)) {
         status->code = 1;
         LOG(WARNING) << "get insert information failed";
         return false;
     }
-    std::shared_ptr<SQLInsertRow> row = std::make_shared<SQLInsertRow>(
-        table_info, ::openmldb::sdk::ConvertToSchema(table_info), default_map, str_length);
-    if (!row) {
-        status->msg = "fail to parse row from sql " + sql;
-        LOG(WARNING) << status->msg;
-        return false;
-    }
-    if (!row->Init(0)) {
-        status->msg = "fail to encode row for table " + table_info->name();
+
+    if (default_maps.empty() || str_lengths.empty() || default_maps.size() != str_lengths.size()) {
+        status->msg = "default maps is empty or str_lengths are empty or default_maps";
+        status->code = 1;
         LOG(WARNING) << status->msg;
         return false;
     }
-    if (!row->IsComplete()) {
-        status->msg = "insert value isn't complete";
+    if (default_maps.size() != str_lengths.size()) {
```

Comment:
Perhaps this check should be put in the ```GetMultiRowInsertInfo``` function, since they both come from that funciton.

Reply:
Fixed